packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -1372,8 +1381,9 @@ export class Cluster extends ClusterBase {
     this.prune = props.prune ?? true;
     this.vpc = props.vpc || new ec2.Vpc(this, 'DefaultVpc');
 
-    if (props.version === KubernetesVersion.V1_22 && !props.kubectlLayer) {
-      Annotations.of(this).addWarning(`You created a cluster with Kubernetes Version ${props.version} without specifying the kubectlLayer property. This may cause failures as the kubectl version provided with aws-cdk-lib is 1.20, which is only guaranteed to be compatible with Kubernetes versions 1.19-1.21. Please provide a kubectlLayer from @aws-cdk/lambda-layer-kubectl-v22.`);
+    const kubectlVersion = Number(props.version.version) * 100;
```

Comment:
I feel we should just use `semver` here to compare versions, no need to parse to a number and *100 stuff.

Reply:
Okay fair enough, I'm fine with either approach. We can update this.

Reply:
it's already a dependency of aws-cdk-lib so no cost really. Otherwise we will have to parse each individual segment major + minor from both and compare both but since semver already does this that just feels nicer.

Reply:
Updated 👍